utils/build-script

```diff
@@ -147,6 +156,26 @@ def print_xcodebuild_versions(file=sys.stdout):
     print(fmt.format(version=version, sdks=sdks), file=file)
     file.flush()
 
+def validate_xcode_compatibility():
+    if sys.platform != 'darwin':
+        return
+
+    if os.getenv("SKIP_XCODE_VERSION_CHECK"):
+        print("note: skipping Xcode version check")
+        return
+
+    version = shell.capture(
+        ['xcodebuild', '-version'], dry_run=False, echo=False).strip()
+
+    valid_build_numbers = tuple(x[1] for x in _SUPPORTED_XCODE_BUILDS)
+    if not version.endswith(valid_build_numbers):
+        valid_versions_string = "\n".join(
+            "{} ({})".format(*x) for x in _SUPPORTED_XCODE_BUILDS)
+        raise SystemExit(
+            "error: using unsupported Xcode version:\n\n{}\n\nInstall one of:\n{}\n\nOr set 'SKIP_XCODE_VERSION_CHECK=1' in the environment".format(
```

Comment:
I think this should be a warning, because if we are working with future Xcode versions internally it will be pain to keep it up to date.

Reply:
Looks like this was discussed a bit on https://bugs.swift.org/browse/SR-13497

but my 2cents:

- Not making this block means no one will ever see it because the build script has too much output
- This probably won't be nearly as helpful for Apple people as external people since I assume you're juggling a lot of different daily Xcode versions

Reply:
+1 to making this opt out instead of opt in

Reply:
Sounds good, we can have this disabled by default internally.